mne/beamformer/tests/test_external.py

```diff
@@ -0,0 +1,93 @@
+# preparing everything:
+import mne
+import numpy as np
+import os.path as op
+from scipy.io import savemat
+
+from mne.datasets import testing
+from mne.beamformer import make_lcmv, apply_lcmv
+from mne.beamformer.tests.test_lcmv import _get_data
+
+
+data_path = testing.data_path(download=False)
+fname_raw = op.join(data_path, 'MEG', 'sample', 'sample_audvis_trunc_raw.fif')
+fname_cov = op.join(data_path, 'MEG', 'sample', 'sample_audvis_trunc-cov.fif')
+fname_fwd = op.join(data_path, 'MEG', 'sample',
+                    'sample_audvis_trunc-meg-eeg-oct-4-fwd.fif')
+fname_fwd_vol = op.join(data_path, 'MEG', 'sample',
+                        'sample_audvis_trunc-meg-vol-7-fwd.fif')
+fname_event = op.join(data_path, 'MEG', 'sample',
+                      'sample_audvis_trunc_raw-eve.fif')
+fname_label = op.join(data_path, 'MEG', 'sample', 'labels', 'Aud-lh.label')
+
+reject = dict(grad=4000e-13, mag=4e-12)
+
+# where FieldTrip data should be saved to / imported from
+ft_data_path = ' '
+
+
+def _get_bf_data(save_fieldtrip=False):
+    raw, epochs, evoked, data_cov, _, _, _, _, _, fwd = _get_data(proj=False)
+
+    if save_fieldtrip is True:
+        # raw needs to be saved with all channels and picked in FieldTrip
+        raw.save(op.join(ft_data_path, 'raw.fif'), overwrite=True)
+
+        # src (tris are not available in fwd['src'] once imported into MATLAB)
+        src = fwd['src'].copy()
+        mne.write_source_spaces(op.join(ft_data_path, 'src.fif'), src)
+
+    # pick gradiometers only:
+    epochs.pick_types(meg='grad')
+    evoked.pick_types(meg='grad')
+
+    # compute covariance matrix
+    data_cov = mne.compute_covariance(epochs, tmin=0.04, tmax=0.145,
+                                      method='empirical')
+
+    if save_fieldtrip is True:
+        # if the covariance matrix and epochs need resaving:
+        # data covariance:
+        cov_savepath = op.join(ft_data_path, 'sample_cov')
+        sample_cov = {'sample_cov': data_cov['data']}
+        savemat(cov_savepath, sample_cov)
+        # evoked data:
+        ev_savepath = op.join(ft_data_path, 'sample_evoked')
+        data_ev = {'sample_evoked': evoked.data}
+        savemat(ev_savepath, data_ev)
+
+    return evoked, data_cov, fwd
+
+
+def test_lcmv_fieldtrip():
```

Comment:
Does this need a decorator ``@testing.requires_testing_data`` ?

Reply:
Yes